<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InterfaceDSPIC</class>
 <widget class="QDialog" name="InterfaceDSPIC">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>889</width>
    <height>605</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>InterfaceDSPIC</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="8" column="1">
    <widget class="QGroupBox" name="gpbExperiment">
     <property name="title">
      <string>Experiment</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_9">
      <item row="0" column="1" colspan="2">
       <widget class="QComboBox" name="cmbExperimentName"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="lblExperimentName">
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QPushButton" name="btnStartExperiment">
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" colspan="3">
       <widget class="QLineEdit" name="leExperimentName"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="lblExperimentDataRate">
        <property name="text">
         <string>Data rate [Hz]</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="leExperimentDataRate">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLineEdit" name="leExperimentName_3">
        <property name="minimumSize">
         <size>
          <width>40</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="lblExperimentDuration">
        <property name="text">
         <string>Duration [s]</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="btnNewExperiment">
        <property name="text">
         <string>New</string>
        </property>
       </widget>
      </item>
      <item row="7" column="2" colspan="2">
       <widget class="QPushButton" name="btnStopExperiment">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="btnDeleteExperiment">
        <property name="text">
         <string>Delete</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="4">
       <widget class="QGroupBox" name="gpbMuxInputs">
        <property name="title">
         <string>Multiplexer inputs</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_10">
         <item row="2" column="0">
          <widget class="QCheckBox" name="ckbMuxInput_3">
           <property name="text">
            <string>Input 3</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="ckbMuxInput_7">
           <property name="text">
            <string>Input 7</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="ckbMuxInput_2">
           <property name="text">
            <string>Input 2</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="ckbMuxInput_6">
           <property name="text">
            <string>Input 6</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="ckbMuxInput_1">
           <property name="text">
            <string>Input 1</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="ckbMuxInput_5">
           <property name="text">
            <string>Input 5</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="ckbMuxInput_4">
           <property name="text">
            <string>Input 4</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="ckbMuxInput_8">
           <property name="text">
            <string>Input 8</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" rowspan="8">
    <widget class="QGroupBox" name="gbpConsole">
     <property name="title">
      <string>Console</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="3" column="3">
       <widget class="QPushButton" name="btnClearConsole">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" rowspan="2" colspan="4">
       <widget class="QTextEdit" name="teConsole">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="acceptRichText">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPushButton" name="btnExit">
        <property name="text">
         <string>Exit</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QGroupBox" name="gpbDac">
     <property name="title">
      <string>Dac</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="1" column="0">
       <widget class="QLabel" name="lblPwmFreq">
        <property name="text">
         <string>Period</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="leDacValue">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="lblDacValue">
        <property name="text">
         <string>Value</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="btnPwmGetDuty">
        <property name="text">
         <string>Get</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="btnDacSetValue">
        <property name="text">
         <string>Set</string>
        </property>
       </widget>
      </item>
      <item row="8" column="3">
       <widget class="QPushButton" name="btnPwmWarm">
        <property name="text">
         <string>WarmUp</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <spacer name="vspDac">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="3" column="3">
       <widget class="QPushButton" name="btnPwmSetDuty">
        <property name="text">
         <string>Set</string>
        </property>
       </widget>
      </item>
      <item row="9" column="3">
       <widget class="QPushButton" name="btnPwmRun">
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="btnPwmGetFreq">
        <property name="text">
         <string>Get</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btnDacGetValue">
        <property name="text">
         <string>Get</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="lePwmFreq"/>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="btnPwmSetFreq">
        <property name="text">
         <string>Set</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="lblPwmDuty">
        <property name="text">
         <string>DutyCycle</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="lePwmDuty"/>
      </item>
      <item row="6" column="3">
       <widget class="QPushButton" name="btnPwmReset">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QPushButton" name="btnSquareRun">
        <property name="text">
         <string>Square</string>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QPushButton" name="btnSquareStop">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0" colspan="2">
       <widget class="QSlider" name="hsDacValue">
        <property name="maximum">
         <number>25</number>
        </property>
        <property name="pageStep">
         <number>5</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="sliderPosition">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedControls">
         <bool>false</bool>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
        <property name="tickInterval">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLCDNumber" name="lcdDacValue"/>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="lblDacSlider">
        <property name="text">
         <string>Amplitude </string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="2" rowspan="2">
    <widget class="QGroupBox" name="gpbBCI">
     <property name="title">
      <string>BCI</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="11" column="2">
       <widget class="QPushButton" name="btnStopBCI">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0" colspan="4">
       <widget class="QtCharts::QChartView" name="cvEEG"/>
      </item>
      <item row="11" column="1">
       <widget class="QPushButton" name="btnRunBCI">
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="ckbReadBCI">
        <property name="text">
         <string>Read Brainwave</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="lblAttention">
        <property name="text">
         <string>Attention level</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLCDNumber" name="lcdAttention"/>
      </item>
      <item row="0" column="0" colspan="4">
       <widget class="QGroupBox" name="gpbBCIcomm">
        <property name="title">
         <string>Connection</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="1">
          <widget class="QComboBox" name="cmbSerialBCI"/>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="btnRefreshSerialBCI">
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lblSerialBCI">
           <property name="text">
            <string>Serial Port</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="btnConnectBCI">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="btnDIsconnectBCI">
           <property name="text">
            <string>Disconnect</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QCheckBox" name="ckbDriveBCI">
        <property name="text">
         <string>Control the DAC value</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="gpbCommunication">
     <property name="title">
      <string>Communication</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" colspan="2">
       <widget class="QGroupBox" name="gpbCommunicationConfiguration">
        <property name="title">
         <string>Configuration</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="1">
          <widget class="QComboBox" name="cmbBaudRates"/>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="btnOpenCommunicationWithMicrocontroller">
           <property name="text">
            <string>Open</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="btnRefreshSerialNames">
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="btnCloseCommunicationWithMicrocontroller">
           <property name="text">
            <string>Close</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lblSerialNames">
           <property name="text">
            <string>Serial Port</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="cmbSerialNames"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lblBaudRates">
           <property name="text">
            <string>Baudrate (bps)</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="leMessageToSend">
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="btnSendMessage">
        <property name="text">
         <string>Send</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="vlyMux">
     <property name="title">
      <string>Multiplexer</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_14">
      <item row="1" column="1">
       <widget class="QRadioButton" name="radMuxInput2">
        <property name="text">
         <string>Input 2</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="radMuxNoInput">
        <property name="text">
         <string>No input</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="radMuxInput1">
        <property name="text">
         <string>Input 1</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QRadioButton" name="radMuxInput3">
        <property name="text">
         <string>Input 3</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="radMuxInput5">
        <property name="text">
         <string>Input 5</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QRadioButton" name="radMuxInput7">
        <property name="text">
         <string>Input 7</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QRadioButton" name="radMuxInput8">
        <property name="text">
         <string>Input 8</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QRadioButton" name="radMuxInput6">
        <property name="text">
         <string>Input 6</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QRadioButton" name="radMuxInput4">
        <property name="text">
         <string>Input 4</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">btgMux</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="2">
    <widget class="QGroupBox" name="gpbGrip">
     <property name="title">
      <string>Grasping experiment</string>
     </property>
     <widget class="QCheckBox" name="ckbGrip">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>22</y>
        <width>92</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Gripping setup</string>
      </property>
     </widget>
     <widget class="QLabel" name="lblGtime">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>44</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Time</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnGrip">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>70</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdGtime">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>40</y>
        <width>151</width>
        <height>23</height>
       </rect>
      </property>
      <property name="digitCount">
       <number>9</number>
      </property>
     </widget>
     <widget class="QPushButton" name="btnGripStop">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>70</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>PushButton</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="tbGrip">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QtCharts::QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">QtCharts/QChartView&gt;
#include &lt;QtCharts/chartsnamespace.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="btgMux"/>
 </buttongroups>
</ui>
